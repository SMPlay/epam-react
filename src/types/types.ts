export type SearchType = "title" | "genres";
export type SortType = "rating" | "release-date";
