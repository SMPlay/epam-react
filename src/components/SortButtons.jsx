import React from "react";
import PropTypes from "prop-types";
import { createUseStyles } from "react-jss";
import { Button } from "./Button";

const useStyles = createUseStyles({
  sortButtonsContainer: {
    display: "flex",
  },
  sortButtons: {
    width: "165px",
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
  },
  sortButtonsTitle: {
    marginRight: "10px",
  },
});

export const SortButtons = (props) => {
  const classes = useStyles();
  const { onClick, sortBy } = props;

  return (
    <div className={classes.sortButtonsContainer}>
      <p className={classes.sortButtonsTitle}>Sort by</p>
      <Button
        type="secondary"
        onClick={() => onClick("release-date")}
        active={sortBy === "release-date"}
        fontSize="15px"
      >
        {" "}
        release date

      </Button>
      <Button
        type="secondary"
        onClick={() => onClick("rating")}
        active={sortBy === "rating"}
        fontSize="15px"
      >
        {" "}
        rating

      </Button>
    </div>
  );
};

SortButtons.propTypes = {
  onClick: PropTypes.func.isRequired,
  sortBy: PropTypes.string.isRequired,
};
