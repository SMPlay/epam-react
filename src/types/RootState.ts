import { Map } from "immutable";

import { Film } from "./Film";
import { SearchType, SortType } from "./types";

export interface RootState {
  filmsData: Film[] | [];
  searchBy: SearchType;
  sortBy: SortType;
  searchInputValue: string;
  currentFilm: Film | {};
  isLoading: boolean;
}
