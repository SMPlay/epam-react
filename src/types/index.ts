export { Film } from "./Film";
export { SearchType, SortType } from "./types";
