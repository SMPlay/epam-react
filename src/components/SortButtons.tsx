import React from "react";
import { createUseStyles } from "react-jss";
import { Button } from "./Button";
import { SortType } from "../types";

const useStyles = createUseStyles({
  sortButtonsContainer: {
    display: "flex",
  },
  sortButtons: {
    width: "165px",
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
  },
  sortButtonsTitle: {
    marginRight: "10px",
  },
});

interface Props {
  onClick: (sortBy: SortType) => void;
  sortBy: SortType;
}

export const SortButtons: React.FC<Props> = (props: Props) => {
  const classes = useStyles();
  const { onClick, sortBy } = props;

  return (
    <div className={classes.sortButtonsContainer}>
      <p className={classes.sortButtonsTitle}>Sort by</p>
      <Button
        type="secondary"
        onClick={() => onClick("release-date")}
        active={sortBy === "release-date"}
        fontSize="15px"
      >
        &nbsp;release date
      </Button>
      <Button
        type="secondary"
        onClick={() => onClick("rating")}
        active={sortBy === "rating"}
        fontSize="15px"
      >
        &nbsp;rating
      </Button>
    </div>
  );
};
