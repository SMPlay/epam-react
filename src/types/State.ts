import { Film } from "./Film";
import { SearchType, SortType } from "./types";

export interface State {
  filmsData: Film[];
  searchBy: SearchType;
  sortBy: SortType;
  searchInputValue: string;
  currentFilm: Film | {};
  isLoading: boolean;
}
