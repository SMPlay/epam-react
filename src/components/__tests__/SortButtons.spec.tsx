import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";
import { SortButtons } from "../SortButtons";
import { SortType } from "../../types";

describe("SortButtons component", () => {
  interface Props {
    onClick: (sortBy: SortType) => void;
    sortBy: SortType;
  }

  let props: Props;
  let component = mount<Router>(
    <Router>
      <SortButtons {...props} />
    </Router>,
  );
  beforeEach(() => {
    props = {
      onClick: jest.fn(),
      sortBy: "rating",
    };
  });
  afterEach(() => {
    component = mount<Router>(
      <Router>
        <SortButtons {...props} />
      </Router>,
    );
  });

  it("should render with props", () => {
    const tree = renderer.create(
      <Router>
        <SortButtons {...props} />
      </Router>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
