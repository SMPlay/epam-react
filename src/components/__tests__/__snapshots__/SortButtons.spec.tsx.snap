// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortButtons component should render with props 1`] = `
<div
  className="sortButtonsContainer-0-2-1"
>
  <p
    className="sortButtonsTitle-0-2-3"
  >
    Sort by
  </p>
  <button
    className="btn-0-2-4 btn-d0-0-2-7 btn_secondary-0-2-6 "
    data-test-button="sort-by-release-date"
    onClick={[Function]}
  >
     release date
  </button>
  <button
    className="btn-0-2-4 btn-d2-0-2-9 btn_secondary-0-2-6 active"
    data-test-button="sort-by-rating"
    onClick={[Function]}
  >
     rating
  </button>
</div>
`;
